test/src/test/java/org/corfudb/runtime/collections/PersistentCorfuTableTest.java

```diff
@@ -1263,4 +1263,89 @@ public void testNestedIndexesWithNullValues() throws Exception {
         assertThat(entries.size()).isEqualTo(2);
         rt.getObjectsView().TXEnd();
     }
+
+    /**
+     * Test that a table without any updates can be served when a concurrent
+     * transaction syncs the stream forward before this first transaction has
+     * a chance to request a snapshot proxy.
+     */
+    @Test
+    public void testTableNoUpdateInterleave() {
+        PersistentCorfuTable<String, String>
```

Comment:
This pattern is indeed impacted by caching. In this test, the empty version is not served by `MultiVersionObject:L146` since the global tail is not equal to -1; this is why there is a second table with updates. This test would fail with a TrimmedException previously because the main thread would "miss" its version from `currentObject` as the second thread performed a sync, and the -1 version was never added to the cache.